website/content/docs/pipelines/index.mdx

```diff
@@ -0,0 +1,300 @@
+---
+layout: docs
+page_title: Pipelines
+description: |-
+  The overview of Waypoint Pipelines.
+---
+
+# Waypoint Custom Pipelines
+
+~> **NOTE**: Custom Pipelines is currently a **Tech Preview** feature of Waypoint.
+Given that, we invite all Waypoint users to give custom pipelines a try and let
+us know your experience with it, feature requests and enhancements, or any bugs
+you might find via our [Discuss forums](https://discuss.hashicorp.com/c/waypoint/51)
+or [GitHub](https://github.com/hashicorp/waypoint/issues). We look forward to
+hearing about your experience and ways Waypoint can help make your experience
+delivering your applications even better!
+
+Waypoint before the introduction of Pipelines has always had a relatively fixed
+built-in pipeline with the `waypoint up` CLI which ran a build, deploy, and
+release. An "Up" was a special operation that encapsulated a Build, Deploy, and
+Release for an application. For most users getting started with Waypoint, this
+simple pipeline is powerful enough to get off the ground continuously building
+and deploying an application to production. However, as projects and teams
+become more complicated addressing real world concerns, users wished for more
+control and customization over the order in which Waypoint deploys applications.
+Running tests after a build, security scanning artifacts prior to a deploy, and
+so on. This was not easily possible with the current Waypoint model. Custom
+Pipelines are an answer to this problem. With this tech preview, users can
+define pipelines that can run in whichever order required to deliver an
+application. Additionally, pipelines allow for custom execution steps through
+the "Exec" plugin that will run any kind of bash script as part of the pipeline
+execution for operations like running tests or security scanning.
+
+Pipelines give users a way to tell Waypoint what order these operations should
+be run in, as well as any operation options that might need to be configured
+like disabling an artifact push for the build step, or pruning old deployments
+on a release in the release step. Waypoint projects still expect the
+application to be defined outside of a pipeline in its own `app` stanza.
```

Comment:
Minor edits, plus removed a sentence that reads more like a "how to use pipelines" and less like a "marketing summary of why pipelines is cool." That sentence can move to the how-to part.
```suggestion
Pipelines allow users to tell Waypoint what order these operations should
run in and configure any operation options necessary, like disabling an
artifact push in the `build` step, or pruning old deployments
on a release in the `release` step.
```